demo/demo.h

```diff
@@ -37,37 +44,207 @@ DECLARE_int32(demo_port);
+          "{}\n",
+          HTTPResponseCode::OK, "application/json");
+    });
+    HTTP(port).Register("/layout/data", [](Request r) {
+      std::thread([](Request&& r) {
+                    // Since we are in another thread, need to catch exceptions ourselves.
+                    try {
+                      auto response = r.connection.SendChunkedHTTPResponse(
+                          HTTPResponseCode::OK,
+                          "application/json; charset=utf-8",
+                          {{"Connection", "keep-alive"}, {"Access-Control-Allow-Origin", "*"}});
+                      std::string data;
+                      const double begin = static_cast<double>(Now());
+                      const double t = atof(r.url.query["t"].c_str());
+                      const double end = (t > 0) ? (begin + t * 1e3) : 1e18;
+                      double current;
+                      while ((current = static_cast<double>(Now())) < end) {
+                        std::this_thread::sleep_for(std::chrono::milliseconds(rand() % 100 + 100));
+                        const double x = current;
+                        const double y = sin(5e-3 * (current - begin));
+                        data += Printf("{\"x\":%lf,\"y\":%lf}\n", x, y);
+                        const double f = (rand() % 101) * (rand() % 101) * (rand() % 101) * 1e-6;
+                        const size_t n = static_cast<size_t>(data.length() * f);
+                        if (n) {
+                          response.Send(data.substr(0, n));
+                          data = data.substr(n);
+                        }
+                      }
+                    } catch (const std::exception& e) {
+                      std::cerr << "Exception in data serving thread: " << e.what() << std::endl;
+                    }
+                  },
+                  std::move(r)).detach();
+    });
+    HTTP(port).Register("/layout/meta", [](Request r) {
+      r.connection.SendHTTPResponse(ExampleMeta(),
+                                    "meta",
+                                    HTTPResponseCode::OK,
+                                    "application/json; charset=utf-8",
+                                    {{"Connection", "close"}, {"Access-Control-Allow-Origin", "*"}});
+    });
+    HTTP(port).Register("/layout", [](Request r) {
+      LayoutItem layout;
+      LayoutItem row;
+      layout.col.push_back(row);
+      r.connection.SendHTTPResponse(layout,
+                                    "layout",
+                                    HTTPResponseCode::OK,
+                                    "application/json; charset=utf-8",
+                                    {{"Connection", "close"}, {"Access-Control-Allow-Origin", "*"}});
+    });
+    // The "./static/" directory should be a symlink to "Web/build" or "Web/build-dev".
+    FileSystem::ScanDir("./static/", [&port](const std::string& filename) {
+      std::string filepath = "./static/" + filename;
```

Comment:
Nit, here and in a few other places: it's a good habit to prepend immutable variables with ```const```: ```const std::string filepath = ```.